tests/test_repocard.py

```diff
@@ -240,6 +241,22 @@ def test_update_existing_result_with_overwrite(self):
         updated_metadata = metadata_load(self.repo_path / self.REPO_NAME / "README.md")
         self.assertDictEqual(updated_metadata, new_metadata)
 
+    def test_metadata_update_upstream(self):
+        new_metadata = copy.deepcopy(self.existing_metadata)
+        new_metadata["model-index"][0]["results"][0]["metrics"][0]["value"] = 0.1
+
+        path = hf_hub_download(
+            f"{USER}/{self.REPO_NAME}",
+            filename=REPOCARD_NAME,
+            use_auth_token=self._token,
+        )
+
+        metadata_update(
+            f"{USER}/{self.REPO_NAME}", new_metadata, token=self._token, overwrite=True
+        )
+
+        self.assertNotEqual(metadata_load(path), new_metadata)
```

Comment:
can we also check the original file is not touched?

Reply:
Handled in [5fabdf8](https://github.com/huggingface/huggingface_hub/pull/891/commits/5fabdf86b27b75da6704d2c8dffb16c3814e2557)